src/main/kotlin/org/jetbrains/research/testspark/tools/strategies/TestProcessorStrategies.kt

```diff
@@ -0,0 +1,18 @@
+package org.jetbrains.research.testspark.tools.strategies
+
+import org.jetbrains.research.testspark.core.utils.DataFilesUtil
+import java.io.File
+
+class TestProcessorStrategies {
+    companion object {
+        fun getRunner(homeDirectory: String): File {
+            return File(homeDirectory).walk()
+                .filter {
+                    val isJavaName =
+                        if (DataFilesUtil.isWindows()) it.name.equals("java.exe") else it.name.equals("java")
+                    isJavaName && it.isFile
+                }
+                .first()
+        }
+    }
+}
```

Comment:
I suggest making this method a normal function. It just searches for a java compiler; it is not a strategy:

```kotlin
/**
* Finds 'javac' compiler (both on Unix & Windows)
* starting from the provided directory.
*/
 fun findJavaCompilerInDirectory(homeDirectory: String): File {
            return File(homeDirectory).walk()
                .filter {
                    val isJavaName =
                        if (DataFilesUtil.isWindows()) it.name.equals("java.exe") else it.name.equals("java")
                    isJavaName && it.isFile
                }
                .first()
        }
```